shell/browser/ui/views/opaque_frame_view.h

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 Microsoft GmbH.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ELECTRON_SHELL_BROWSER_UI_VIEWS_OPAQUE_FRAME_VIEW_H_
+#define ELECTRON_SHELL_BROWSER_UI_VIEWS_OPAQUE_FRAME_VIEW_H_
+
+#include <memory>
+
+#include "base/memory/raw_ptr.h"
+#include "base/scoped_observation.h"
+#include "chrome/browser/ui/view_ids.h"
+#include "shell/browser/ui/views/frameless_view.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/gfx/font_list.h"
+#include "ui/linux/nav_button_provider.h"
+#include "ui/linux/window_button_order_observer.h"
+#include "ui/views/controls/button/button.h"
+#include "ui/views/window/caption_button_types.h"
+#include "ui/views/window/frame_buttons.h"
+#include "ui/views/window/non_client_view.h"
+
+class CaptionButtonPlaceholderContainer;
+
+namespace electron {
+
+class OpaqueFrameView : public FramelessView {
+  METADATA_HEADER(OpaqueFrameView, FramelessView)
+
+ public:
+  // Constants used by OpaqueBrowserFrameView as well.
+  static const int kContentEdgeShadowThickness;
+
+  // The frame border is only visible in restored mode and is hardcoded to 4
+  // px on each side regardless of the system window border size.  This is
+  // overridable by subclasses, so RestoredFrameBorderInsets() should be used
+  // instead of using this constant directly.
+  static constexpr int kFrameBorderThickness = 4;
```

Comment:
This causes a think frame border for resizing outside the client area in window restored mode, also this seems chrome UI specific. Can we remove it ?